sql/service/if/TCLIService.thrift

```diff
@@ -38,34 +38,7 @@ namespace cpp apache.spark.sql.service.rpc.thrift
 // List of protocol versions. A new token should be
 // added to the end of this list every time a change is made.
 enum TProtocolVersion {
-  HIVE_CLI_SERVICE_PROTOCOL_V1,
-
-  // V2 adds support for asynchronous execution
-  HIVE_CLI_SERVICE_PROTOCOL_V2
-
-  // V3 add varchar type, primitive type qualifiers
-  HIVE_CLI_SERVICE_PROTOCOL_V3
-
-  // V4 add decimal precision/scale, char type
-  HIVE_CLI_SERVICE_PROTOCOL_V4
-
-  // V5 adds error details when GetOperationStatus returns in error state
-  HIVE_CLI_SERVICE_PROTOCOL_V5
-
-  // V6 uses binary type for binary payload (was string) and uses columnar result set
-  HIVE_CLI_SERVICE_PROTOCOL_V6
-
-  // V7 adds support for delegation token based connection
-  HIVE_CLI_SERVICE_PROTOCOL_V7
-
-  // V8 adds support for interval types
-  HIVE_CLI_SERVICE_PROTOCOL_V8
-
-  // V9 adds support for serializing ResultSets in SerDe
-  HIVE_CLI_SERVICE_PROTOCOL_V9
-
-  // V10 adds support for in place updates via GetOperationStatus
-  HIVE_CLI_SERVICE_PROTOCOL_V10
+  SPARK_SQL_SERVICE_PROTOCOL_V1,
```

Comment:
cc @juliuszsompolski

Reply:
If this is to remain compatible with drivers - both open source Hive JDBC, as well as proprietary Simba, Progress, CData drivers, we need to keep supporting all these Hive enum values.
For example Simba JDBC works with HIVE_CLI_SERVICE_PROTOCOL_V6 and Simba ODBC works with HIVE_CLI_SERVICE_PROTOCOL_V8.
We cannot drop support for these, as they are installed by clients (e.g. Tableau works with Simba ODBC installed on the system) or embedded in BI products (e.g. PowerBI embeds Simba ODBC drivers).
Since the wire protocol is still the "Hive" protocol, I wouldn't even rename it either.

Reply:
Maybe this is our only chance to build our own Thrift protocol, [JDBC driver](https://github.com/spark-thriftserver/spark-thriftserver/tree/master/sql/jdbc) and ODBC driver.  I agree to still use Hive's Thrift protocol because it is widely used:
https://github.com/apache/impala/blob/master/common/thrift/hive-1-api/TCLIService.thrift
https://github.com/apache/arrow/blob/master/cpp/src/arrow/dbi/hiveserver2/thrift/TCLIService.thrift

Reply:
cc @gatorsmile @AngersZhuuuu

Reply:
I agree we should make our JDBC/ODBC server / drivers independent of Hive, but it's necessary to keep backwards compatibility.
We cannot remove, renumber or change the behaviour of any existing fields.

If we want to make some additions, we would have to introduce a new TProtocolVersion value >> than all the Hive values, and then check the version negotiated in OpenSession, feature flagging the additions, and bumping that version number every time we change anything.
It adds a new dimension of maintenance and backwards/forwards compatibility testing, so I would only do it when we want to make some substantial improvements.
I think that just adding INTERVAL in the other PR isn't enough to fork this new version now.

Reply:
OK. I reverted these changes.